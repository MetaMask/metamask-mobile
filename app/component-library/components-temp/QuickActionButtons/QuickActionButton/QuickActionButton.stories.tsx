/* eslint-disable no-console */
// Third party dependencies
import React from 'react';

// External dependencies
import {
  Box,
  ButtonSize,
  ButtonVariant,
} from '@metamask/design-system-react-native';

// Internal dependencies
import QuickActionButton from './QuickActionButton';
import { QuickActionButtonProps } from './QuickActionButton.types';

const QuickActionButtonMeta = {
  title: 'Components Temp / QuickActionButtons / QuickActionButton',
  component: QuickActionButton,
  argTypes: {
    children: {
      control: { type: 'text' },
      description: 'The content to display inside the button',
    },
    isDisabled: {
      control: { type: 'boolean' },
      description: 'Whether the button is disabled',
    },
  },
  args: {
    children: '25%',
    onPress: () => console.log('Button pressed'),
  },
};

export default QuickActionButtonMeta;

export const Default = {};

export const Size = {
  render: function Render(args: QuickActionButtonProps) {
    return (
      <Box gap={4}>
        <QuickActionButton {...args} size={ButtonSize.Sm} />
        <QuickActionButton {...args} size={ButtonSize.Md} />
        <QuickActionButton {...args} size={ButtonSize.Lg} />
      </Box>
    );
  },
};

export const IsDisabled = {
  args: {
    isDisabled: true,
  },
};

export const Variant = {
  render: function Render(args: QuickActionButtonProps) {
    return (
      <Box gap={4}>
        <QuickActionButton {...args} variant={ButtonVariant.Primary} />
        <QuickActionButton {...args} variant={ButtonVariant.Secondary} />
        <QuickActionButton {...args} variant={ButtonVariant.Tertiary} />
      </Box>
    );
  },
};
