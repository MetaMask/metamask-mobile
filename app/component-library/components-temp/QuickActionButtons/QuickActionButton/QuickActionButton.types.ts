// Third party dependencies
import { ReactNode } from 'react';

// External dependencies
import {
  ButtonProps,
  ButtonVariant,
} from '@metamask/design-system-react-native';

/**
 * Individual QuickActionButton component props
 */
export interface QuickActionButtonProps
  extends Omit<ButtonProps, 'children' | 'variant'> {
  /**
   * The content to display inside the button
   */
  children: ReactNode;
  /**
   * Callback when the button is pressed
   */
  onPress: () => void;
  /**
   * Whether the button is disabled
   * @default false
   */
  isDisabled?: boolean;
  /**
   * Button variant
   * @default ButtonVariant.Secondary
   */
  variant?: ButtonVariant;
}
