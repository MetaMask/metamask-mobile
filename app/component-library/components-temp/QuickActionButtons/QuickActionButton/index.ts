export { default } from './QuickActionButton';
export type { QuickActionButtonProps } from './QuickActionButton.types';
