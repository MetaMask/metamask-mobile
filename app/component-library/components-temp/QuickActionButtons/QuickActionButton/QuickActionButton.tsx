// Third party dependencies
import React from 'react';

// External dependencies
import {
  Button,
  ButtonVariant,
  ButtonSize,
} from '@metamask/design-system-react-native';

// Internal dependencies
import { QuickActionButtonProps } from './QuickActionButton.types';

/**
 * Individual QuickActionButton component
 */
const QuickActionButton: React.FC<QuickActionButtonProps> = ({
  children,
  onPress,
  isDisabled = false,
  variant = ButtonVariant.Secondary,
  size = ButtonSize.Lg,
  testID,
  ...props
}) => (
  <Button
    variant={variant}
    size={size}
    testID={testID}
    onPress={onPress}
    isDisabled={isDisabled}
    isFullWidth
    {...props}
    twClassName={`px-2 min-w-0 ${props.twClassName || ''}`}
  >
    {children}
  </Button>
);

export default QuickActionButton;
