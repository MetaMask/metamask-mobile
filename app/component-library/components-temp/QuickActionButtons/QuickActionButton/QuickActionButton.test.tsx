// Third party dependencies.
import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';

// Internal dependencies.
import QuickActionButton from './QuickActionButton';
import {
  ButtonVariant,
  ButtonSize,
} from '@metamask/design-system-react-native';

describe('QuickActionButton', () => {
  const mockOnPress = jest.fn();

  beforeEach(() => {
    mockOnPress.mockClear();
  });

  it('renders correctly with default props', () => {
    const { toJSON, getByText } = render(
      <QuickActionButton onPress={mockOnPress}>Test Button</QuickActionButton>,
    );

    expect(getByText('Test Button')).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly with custom variant and size', () => {
    const { toJSON } = render(
      <QuickActionButton
        onPress={mockOnPress}
        variant={ButtonVariant.Primary}
        size={ButtonSize.Md}
      >
        Primary Button
      </QuickActionButton>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly in disabled state', () => {
    const { toJSON, getByText } = render(
      <QuickActionButton onPress={mockOnPress} isDisabled>
        Disabled Button
      </QuickActionButton>,
    );

    expect(getByText('Disabled Button')).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it('calls onPress handler when pressed', () => {
    const { getByText } = render(
      <QuickActionButton onPress={mockOnPress}>
        Clickable Button
      </QuickActionButton>,
    );

    const button = getByText('Clickable Button');
    fireEvent.press(button);

    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });

  it('does not call onPress when disabled', () => {
    const { getByText } = render(
      <QuickActionButton onPress={mockOnPress} isDisabled>
        Disabled Button
      </QuickActionButton>,
    );

    const button = getByText('Disabled Button');
    fireEvent.press(button);

    expect(mockOnPress).not.toHaveBeenCalled();
  });

  it('applies custom testID', () => {
    const { getByTestId } = render(
      <QuickActionButton onPress={mockOnPress} testID="custom-button">
        Test Button
      </QuickActionButton>,
    );

    expect(getByTestId('custom-button')).toBeTruthy();
  });
});
