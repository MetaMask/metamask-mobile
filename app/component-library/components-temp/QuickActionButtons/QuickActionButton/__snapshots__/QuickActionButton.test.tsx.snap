// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuickActionButton renders correctly in disabled state 1`] = `
<View
  style={
    [
      {
        "transform": [
          {
            "scale": 1,
          },
        ],
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    accessibilityLabel="Disabled Button"
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#3c4d9d0f",
          "borderColor": "transparent",
          "borderRadius": 12,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 48,
          "justifyContent": "center",
          "minWidth": 0,
          "opacity": 0.5,
          "overflow": "hidden",
          "paddingLeft": 8,
          "paddingRight": 8,
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "bottom": 0,
          "display": "flex",
          "justifyContent": "center",
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      testID="spinner-container"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "columnGap": 8,
              "flexDirection": "row",
            },
            undefined,
          ]
        }
        testID="spinner"
      >
        <View
          style={
            [
              {
                "transform": [
                  {
                    "rotate": "0deg",
                  },
                ],
              },
            ]
          }
          testID="spinner-animated-view"
        >
          <SvgMock
            fill="currentColor"
            name="Loading"
            style={
              [
                {
                  "color": "#121314",
                  "height": 20,
                  "width": 20,
                },
                undefined,
              ]
            }
            testID="spinner-icon"
          />
        </View>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "columnGap": 8,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID="content-container"
    >
      <Text
        accessibilityRole="text"
        ellipsizeMode="clip"
        numberOfLines={1}
        style={
          [
            {
              "color": "#121314",
              "flexGrow": 0,
              "flexShrink": 1,
              "flexWrap": "wrap",
              "fontFamily": "Geist Medium",
              "fontSize": 16,
              "fontWeight": 400,
              "letterSpacing": 0,
              "lineHeight": 24,
              "textAlign": "center",
            },
            undefined,
          ]
        }
      >
        Disabled Button
      </Text>
    </View>
  </View>
</View>
`;

exports[`QuickActionButton renders correctly with custom variant and size 1`] = `
<View
  style={
    [
      {
        "transform": [
          {
            "scale": 1,
          },
        ],
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    accessibilityLabel="Primary Button"
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#121314",
          "borderRadius": 12,
          "flexDirection": "row",
          "height": 40,
          "justifyContent": "center",
          "minWidth": 0,
          "opacity": 1,
          "overflow": "hidden",
          "paddingLeft": 8,
          "paddingRight": 8,
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "bottom": 0,
          "display": "flex",
          "justifyContent": "center",
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      testID="spinner-container"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "columnGap": 8,
              "flexDirection": "row",
            },
            undefined,
          ]
        }
        testID="spinner"
      >
        <View
          style={
            [
              {
                "transform": [
                  {
                    "rotate": "0deg",
                  },
                ],
              },
            ]
          }
          testID="spinner-animated-view"
        >
          <SvgMock
            fill="currentColor"
            name="Loading"
            style={
              [
                {
                  "color": "#ffffff",
                  "height": 20,
                  "width": 20,
                },
                undefined,
              ]
            }
            testID="spinner-icon"
          />
        </View>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "columnGap": 8,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID="content-container"
    >
      <Text
        accessibilityRole="text"
        ellipsizeMode="clip"
        numberOfLines={1}
        style={
          [
            {
              "color": "#ffffff",
              "flexGrow": 0,
              "flexShrink": 1,
              "flexWrap": "wrap",
              "fontFamily": "Geist Medium",
              "fontSize": 16,
              "fontWeight": 400,
              "letterSpacing": 0,
              "lineHeight": 24,
              "textAlign": "center",
            },
            undefined,
          ]
        }
      >
        Primary Button
      </Text>
    </View>
  </View>
</View>
`;

exports[`QuickActionButton renders correctly with default props 1`] = `
<View
  style={
    [
      {
        "transform": [
          {
            "scale": 1,
          },
        ],
      },
      {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    accessibilityLabel="Test Button"
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#3c4d9d0f",
          "borderColor": "transparent",
          "borderRadius": 12,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 48,
          "justifyContent": "center",
          "minWidth": 0,
          "opacity": 1,
          "overflow": "hidden",
          "paddingLeft": 8,
          "paddingRight": 8,
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "bottom": 0,
          "display": "flex",
          "justifyContent": "center",
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      testID="spinner-container"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "columnGap": 8,
              "flexDirection": "row",
            },
            undefined,
          ]
        }
        testID="spinner"
      >
        <View
          style={
            [
              {
                "transform": [
                  {
                    "rotate": "0deg",
                  },
                ],
              },
            ]
          }
          testID="spinner-animated-view"
        >
          <SvgMock
            fill="currentColor"
            name="Loading"
            style={
              [
                {
                  "color": "#121314",
                  "height": 20,
                  "width": 20,
                },
                undefined,
              ]
            }
            testID="spinner-icon"
          />
        </View>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "columnGap": 8,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID="content-container"
    >
      <Text
        accessibilityRole="text"
        ellipsizeMode="clip"
        numberOfLines={1}
        style={
          [
            {
              "color": "#121314",
              "flexGrow": 0,
              "flexShrink": 1,
              "flexWrap": "wrap",
              "fontFamily": "Geist Medium",
              "fontSize": 16,
              "fontWeight": 400,
              "letterSpacing": 0,
              "lineHeight": 24,
              "textAlign": "center",
            },
            undefined,
          ]
        }
      >
        Test Button
      </Text>
    </View>
  </View>
</View>
`;
